mcs/class/System/System.Net/HttpListener.cs

```diff
@@ -348,6 +348,7 @@ void IDisposable.Dispose ()
 			if (disposed)
 				return;
 
+			listening = false;
```

Comment:
Hmm, I'm wondering if this should be in the private `Close` method; `Abort` and (public) `Close` call that as well, and if `Dispose` needs to set listening to false, maybe they do as well?